accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+## Experiences
+
+- Projects using workloads just work if the workloads are installed
+- When opening a project in Visual Studio where the project depends on a workload that is not installed, Visual Studio will use the in-product acquisition experience to pop up a dialog box that notifies the developer what workloads are required, where they can click through to install the workload.
+- When building a project from the command line where the project depends on a workload that is not installed, a friendly error message will be generated specifying which workloads are required, and how to install them (ie by running the VS installer)
+- Improve existing experiences involving SDK resolvers
+  - Improve experience in VS when opening a project with a global.json that specifies an SDK that can't be found
+  - When using the NuGet package MSBuild SDK resolver, the UI should not hang while the NuGet package is acquired
+
+The following may not be delivered in .NET 5, but the design should support it:
+
+- A developer can run `dotnet bundle restore` in order to install all the workloads required by a project and its transitive project references.
+
+## Project files
+
+For .NET 5, iOS, Android, and WPF/Windows Forms project will all use the same `Microsoft.NET.Sdk` MSBuild SDK in the project file.  The `TargetFramework` will be used to specify the operating system, and a project targeting Windows will use the `UseWPF` or `UseWindowsForms` properties to opt in to WPF and/or Windows Forms.  For example, we plan for an Android project to look something like this:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net5.0-android42</TargetFramework>
+  </PropertyGroup>
+
+</Project>
+```
+
+## Workload guidance
+
+The bulk of functionality from workloads will be contained in workload "packs", which will be the unit of delivery for workloads.  This will include MSBuild targets and tasks.  The assets from the packs will be laid out under the dotnet root folder, probably in `packs/<Pack ID>/<Pack Version>` or a similar folder.
+
+Workloads will also provide a manifest package which describes which packs are necessary for which workloads.  The manifest will specify the name and version for each pack that is available for a given .NET SDK release band, as well as which packs are required for each logical workload.  The manifest for workloads will be present even when the workloads it describes are not installed.
+
+### Workload Manifest Targets Files
+
+In addition to the manifest file itself (likely in json format), the workload manifest package will include a `WorkloadManifest.targets` file which should import the necessary .targets files from the workload pack if a project depends on the workload.  The `WorkloadManifest.targets` files from all workload manifests will be imported into all projects that use Microsoft.NET.Sdk.
```

Comment:
Why not call it `Workload.targets`? It seems that the manifest is an implementation detail.

Reply:
It's part of the workload manifest and is imported by all projects.  I think with the name `Workload.targets`, it would be easy to get it confused with the targets which are in the workload and only imported by the workload manifest targets for projects that use the workload.

@mhutch @wli3 @rainersigwald What do you think?  I don't have a strong opinion on the naming here.

Reply:
I prefer longer and descriptive name

WorkloadManifest.targets and Workload.targets both don't really mean too much other than it is related to workload or manifest while everything in the manifest folder is

 We should start from  "a file that should import the necessary .targets files from the workload pack if a project depends on the workload". target  and summarize this sentence to a few words.

WorkloadPackTargetEntry.target?
WorkloadPackTarget.target?
ImportWorkloadPackTaget.target?

Reply:
I get it. The proposal as-is makes sense to me, and I appreciate the explanation.